### PR TITLE
graduate kubewatch to stable

### PR DESCRIPTION
**Pull request by compleatang**
Graduates kubewatch to stable.

### CODE REVIEW

**Comment by mattfarina** [comment]
@compleatang thanks for looking to promote this to stable. We recently documented the process to do so. Could you make sure this PR meets with the guidelines at https://github.com/kubernetes/charts/blob/master/PROCESSES.md#promoting-a-chart-from-incubator-to-stable

**Comment by compleatang** [comment]
@mattfarina as per the other PRs, should comply with the guidelines now

**Comment by compleatang** [comment]
Not sure when Tide pool jobs are triggered, but this should be GTG.

**Comment by unguiculus** [comment]
I'd like you to make a few changes:

* Please apply our revised RBAC best practices: https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md
* Please compare with what `helm create` using Helm 2.8+ gives you and update accordingly.

Thanks.

**Comment by compleatang** [comment]
Thanks @unguiculus as always! 

I think I caught all the best practices around RBAC and new helper functions from helm 2.8 create.

File: incubator/kubewatch/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 name: kubewatch
-version: 0.2.1
+version: 0.2.2
 appVersion: v0.0.3
+deprecated: true  # moved to stable
 description: Kubewatch notifies your slack rooms when changes to your cluster occur
 keywords:
 - kubernetes
```
Comment (unguiculus) [inline incubator/kubewatch/Chart.yaml]: Remove maintainers from deprecated chart.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/kubewatch/Chart.yaml (hunk 1 above)
Comment (compleatang) [reply incubator/kubewatch/Chart.yaml]: done.

**Comment by compleatang** [comment]
not sure what's happening with the CLA bot. but as you can see from my other PR's I've signed it long ago.

File: stable/kubewatch/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+{{ toYaml .Values.resources | indent 10 }}
+    {{- if .Values.affinity }}
+      affinity:
+{{ toYaml .Values.affinity | indent 8 }}
+    {{- end }}
+    {{- if .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml .Values.nodeSelector | indent 8 }}
+    {{- end }}
+      serviceAccountName: {{ if .Values.rbac.create }}{{ template "kubewatch.serviceAccountName" . }}{{ end }}
```
Comment (unguiculus) [inline stable/kubewatch/templates/deployment.yaml]: Remove the `if` block around the service account name.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/kubewatch/templates/deployment.yaml (hunk 2 above)
Comment (compleatang) [reply stable/kubewatch/templates/deployment.yaml]: Done!

File: stable/kubewatch/templates/clusterrole.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{{- if .Values.rbac.create -}}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  labels:
+    app: {{ template "kubewatch.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
```
Comment (unguiculus) [inline stable/kubewatch/templates/clusterrole.yaml]: Please update chart labels: `chart: {{ template "kubewatch.chart" . }}`

**Review by unguiculus — CHANGES_REQUESTED** [review]
One more thing. Sorry.

File: stable/kubewatch/templates/clusterrole.yaml (hunk 3 above)
Comment (compleatang) [reply stable/kubewatch/templates/clusterrole.yaml]: No worries. All done!

**Comment by unguiculus** [comment]
We seem to have a problem with recent changes to the CircleCI job. Will have to see how we can fix this.

**Comment by compleatang** [comment]
Hmm. OK. That does seem to be an issue with the Linter. Let me know if I need to do anything on my end.

**Comment by compleatang** [comment]
Unfortunately I don't have perms to trigger the CircleCI rebuild now that the linting changes have hit. Otherwise this should be GTG @unguiculus

**Comment by unguiculus** [comment]
You need to merge in the latest master to get these changes which in turn will trigger CircleCI.

**Comment by compleatang** [comment]
OK. We're all set now!

**Comment by unguiculus** [comment]
The NOTES.txt of the deprecated chart should also get a deprecation notice.

**Comment by compleatang** [comment]
@unguiculus Done. 

While I'm all for evolving community driven norms, this slow moving of the goal posts on a very unsubstantive PR is rather frustrating. AFAICT I've followed the deprecation requirements as documented in the repo.

**Comment by unguiculus** [comment]
Sorry about that. I can understand your frustration. We just try and do our best here. Feel free to come up with a PR to improve the docs.

**Comment by compleatang** [comment]
Thanks! Yeah I know everyone is doing their best, it's just a bit weird for the goalposts to keep moving. No worries and happy to comply with the docs!

**Comment by compleatang** [comment]
@unguiculus bump here as I think you just need to mark the requested changes were delivered.

**Review by unguiculus — APPROVED** [review]

**Comment by compleatang** [comment]
🎉  (thanks as always @unguiculus for the reviews and patience!)

**Pull request merged**